aten/src/ATen/native/mkldnn/Linear.cpp

```diff
@@ -72,7 +72,10 @@ Tensor mkldnn_linear(
       "mkldnn_linear: input needs to be mkldnn layout");
   if (self.scalar_type() == ScalarType::BFloat16) {
     TORCH_CHECK(mkldnn_bf16_device_check(),
-        "mkldnn_linear: bf16 path needs the cpu support avx512bw, avx512vl and avx512dq");
+        "mkldnn_linear: bf16 path needs the cpu support avx_ne_convert or avx512bw, avx512vl and avx512dq");
+  } else if (self.scalar_type() == ScalarType::Half) {
+    TORCH_CHECK(mkldnn_fp16_device_check(),
+        "mkldnn_linear: fp16 path needs the cpu support avx_ne_convert or avx512_fp16");
```

Comment:
Hmm, is this a correct statement? ARM CPUs support half precision operations. Aren't mkldnn support those?

Reply:
Currently, we are focused on x64. Not sure how complete the support is on ARM. Maybe it can be done in later PRs